classes/SMTPMailingQueueUpdate.php

```diff
@@ -76,4 +79,29 @@ protected function update_1_2_0()
 
 		update_option('smtp_mailing_queue_options', $options);
 	}
+
+	/**
+	 * Due to a bug in 1.3.0 we need to update the stored emails filename.
+	 */
+	protected function update_1_3_1()
+	{
+		$queue = $this->smtpMailingQueue->loadDataFromFiles(true, false);
+		$errors = $this->smtpMailingQueue->loadDataFromFiles(true, true);
+
+		foreach ($queue as $file => $email) {
+			$this->smtpMailingQueue->deleteMail($file);
+			SMTPMailingQueue::storeMail(
+				$email['to'], $email['subject'], $email['message'], $email['headers'],
+				$email['attachments'], $email['time']
+			);
+		}
+
+		foreach ($errors as $file => $email) {
+			$this->smtpMailingQueue->deleteMail($file);
+			SMTPMailingQueue::storeMail(
+				$email['to'], $email['subject'], $email['message'], $email['headers'],
+				$email['attachments'], $email['time'], true
+			);
+		}
+	}
```

Comment:
In fact, filename is only used to uniquely identify emails.
The  name pattern do not have any impact on processing.
I propose to let the old emails with old naming pattern and only start using new pattern for new emails.
In this situation, it is  no more need to given 'invalid' boolean to 'storeMail' method.

Reply:
By default, the `glob` function (used in `loadDataFromFiles`) [sorts the filenames in the alphabetical order](https://www.php.net/manual/en/function.glob.php). So, if some emails are still present in the queue during the upgrade, there is no guarantee that new emails received after the upgrade won't be sent before them... If it is not a relevant issue for you, I am ready to remove the associated code.

Reply:
As uniqid() return distinct format than microtime you are totally right.
Save ordering on upgrade process is not really critical as it is one time process.

Reply:
Fine by me. I remove it.